packages/evolution-frontend/src/components/inputs/InputChoiceSorting.ts

```diff
@@ -0,0 +1,154 @@
+    }
+    out.push(arr.slice(index))
+
+    return out;
+}
+export { verticalByColumns };
+
+const verticalByRows = function (arr: any[], rows: number) {
+    const len = arr.length;
+    const out: any[] = [];
+    let index = 0;
+    while (index + rows < len) {
+        out.push(arr.slice(index, (index += rows)))
+    }
+    out.push(arr.slice(index))
+
+    return out;
+};
+export { verticalByRows };
+
+const horizontalByColumns = function (arr: any[], columns: number ) {
+    const len = arr.length;
+    const out: any[] = [];
+    let index = 0, columnIndex = 0;
+    let newColumn: any[] = [];
+    while(index < len)
+    {
+        newColumn.push(arr[index]);
+        index += columns;
+        if(index >= len && columnIndex < columns) {
+            columnIndex++;
+            index = columnIndex;
+            out.push(newColumn);
+            newColumn = [];
+        }
+    }
+
+    return out;
+}
+export { horizontalByColumns };
+
+const horizontalByRows = function (arr: any[], rows: number) {
+    const len = arr.length;
+    const out: any[] = [];
+    let index = 0, columnIndex = 0, columns = Math.ceil(arr.length / rows);
+    let newColumn: any[] = [];
+    while (index < len) {
+        newColumn.push(arr[index]);
+        index += columns;
+        if(index >= len && columnIndex < columns) {
+            columnIndex++;
+            index = columnIndex;
+            out.push(newColumn);
+            newColumn = [];
+        }
+    }
+
+    return out;
+}
+export { horizontalByRows };
+
+const customVertical = function (arr: any[], custom: number[]) {
+    const len = arr.length;
+    const out: any[] = [];
+    let index = 0, columnIndex = 0;
+    while (index + custom[columnIndex] < len) {
+        out.push(arr.slice(index, index += custom[columnIndex]))
+        columnIndex++;
+    }
+    out.push(arr.slice(index));
+
+    return out;
+}
+export { customVertical }
+
+const customHorizontal = function (arr: any[], custom: number[]) {
+    const len = arr.length, customLen = custom.length;
+    const out: any[] = [];
+    let index = 0, columnIndex = 0, rowIndex = 0;
+    let newColumn: any[] = [];
+    while (index < len && columnIndex < custom[0]) {
+        if (columnIndex >= custom[rowIndex]) {
+            columnIndex++;
+            rowIndex = 0;
+            index = columnIndex;
+            out.push(newColumn);
+            newColumn = [];
+            continue;
+        }
+        newColumn.push(arr[index]);
+        index += custom[rowIndex];
+        rowIndex++;
+        if(index >= len) {
+            columnIndex++;
+            rowIndex = 0;
+            index = columnIndex;
+            out.push(newColumn);
+            newColumn = [];
+        }
+    }
+
+    return out;
+}
+export { customHorizontal };
+
+const _sortByParameters = function (arr: any[], alignment: 'vertical' | 'horizontal' | 'auto', columns?: number, rows?: number, customAlignmentLengths?: number[]) {
```

Comment:
I did not type the input because the functions only reorders the content of the array, what is in the array is not important.

Reply:
In that case, you can make it generic by typing the array to T. This will ensure that callers will receive the same type as what they sent instead of receiving any and not being able to type check anymore